lib/pure/includes/osenv.nim

```diff
@@ -201,33 +100,24 @@ else:
     ## * `existsEnv proc <#existsEnv,string>`_
     ## * `delEnv proc <#delEnv,string>`_
     ## * `envPairs iterator <#envPairs.i>`_
-
-    # Note: by storing the string in the environment sequence,
-    # we guarantee that we don't free the memory before the program
-    # ends (this is needed for POSIX compliance). It is also needed so that
-    # the process itself may access its modified environment variables!
     when nimvm:
       discard "built into the compiler"
     else:
-      var indx = findEnvVar(key)
-      if indx >= 0:
-        environment[indx] = key & '=' & val
-      else:
-        add environment, (key & '=' & val)
-        indx = high(environment)
       when defined(windows) and not defined(nimscript):
         when useWinUnicode:
-          var k = newWideCString(key)
-          var v = newWideCString(val)
-          if setEnvironmentVariableW(k, v) == 0'i32: raiseOSError(osLastError())
+          let k = newWideCString(key)
+          let v = newWideCString(val)
+          if setEnvironmentVariableW(k, v) == 0'i32: 
+            raiseOSError(osLastError(), $(key, val))
         else:
-          if setEnvironmentVariableA(key, val) == 0'i32: raiseOSError(osLastError())
+          if setEnvironmentVariableA(key, val) == 0'i32:
+            raiseOSError(osLastError(), $(key, val))
       elif defined(vcc):
```

Comment:
i don't understand this `elif defined(vcc):`, how can this work?
isn't vcc for windows? if so it'd be taken by the `when defined(windows)` branch above

should be easy to fix

Reply:
I know it works (because it was approved/tested by Araq in my previous PR), so I think this should stay the same? As far as I can tell, `vcc` is a compiler, not a OS. maybe worth asking Araq

Reply:
I tested it but it's possible that a different branch for VCC was taken and I didn't notice. I only tested the runtime aspects, it's quite possible that VCC does support setenv these days.

Reply:
but this is still a bug:
```nim
when defined(windows): ...
elif defined(vcc): ... # this branch cannot be taken on windows
else: ...
```

the correct way should be:
```nim
when defined(vcc): ...
elif defined(windows): ... # ...
else: ...
```

Reply:
Well, if it truly doesn't get taken, but Araq's tests passed, then it should be deleted entirely, right?

Reply:
fixed in https://github.com/nim-lang/Nim/pull/18575